### PR TITLE
CPT: fix PHPCS linting errors in CPT module files, part 2

### PR DESCRIPTION
**Pull request by jeherve**
See #21963

#### Changes proposed in this Pull Request:

* Fix PHPCS errors in Testimonials and in the main module file.

#### Jetpack product discussion

* N/A

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

* Go to Jetpack > Settings > Writing and enable Testimonials
* Go to the new Testimonials menu and create a few Testimonials.
* Go to Posts > Add New, and add a new shortcode as explained [here](https://wordpress.com/support/testimonials/#testimonial-shortcode).
* See how it goes.

### CODE REVIEW

File: projects/plugins/jetpack/modules/custom-post-types/testimonial.php [hunk 1]
```diff
@@ -1,17 +1,28 @@
...
 	const CUSTOM_POST_TYPE       = 'jetpack-testimonial';
 	const OPTION_NAME            = 'jetpack_testimonial';
 	const OPTION_READING_SETTING = 'jetpack_testimonial_posts_per_page';
 
-	public $version = '0.1';
-
-	static function init() {
+	/**
+	 * Initialize class.
+	 */
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/custom-post-types/testimonial.php:20]: May as well document the return type while we're adding doc comments.
```suggestion
	 *
	 * @return self
	 */
```

File: projects/plugins/jetpack/modules/custom-post-types/testimonial.php [hunk 2]
```diff
@@ -185,8 +200,10 @@ function site_supports_custom_post_type() {
 
 	/**
 	 * Add to REST API post type allowed list.
+	 *
+	 *  @param array $post_types Array of post types to add to the allowed list. Default to `array( 'post', 'page', 'revision' )`.
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/custom-post-types/testimonial.php:204]:
```suggestion
	 *  @param array $post_types Array of allowed post types.
	 * @return array `$post_types` with our type added.
```

File: projects/plugins/jetpack/modules/custom-post-types/testimonial.php [hunk 3]
```diff
@@ -283,88 +305,120 @@ function deactivation_post_type_support() {
...
+				'query_var'       => 'testimonial',
+				'show_in_rest'    => true,
+			)
+		);
 	}
 
 	/**
 	 * Update messages for the Testimonial admin.
+	 *
+	 * @param array $messages Existing post update messages.
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/custom-post-types/testimonial.php:365]:
```suggestion
	 * @param array $messages Existing post update messages.
	 * @return array Updated `$messages`.
```

File: projects/plugins/jetpack/modules/custom-post-types/testimonial.php [hunk 4]
```diff
@@ -283,88 +305,120 @@ function deactivation_post_type_support() {
...
+			),
 		);
 
 		return $messages;
 	}
 
 	/**
 	 * Change ‘Enter Title Here’ text for the Testimonial.
+	 *
+	 * @param string $title Placeholder text. Default 'Add title'.
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/custom-post-types/testimonial.php:418]:
```suggestion
	 * @param string $title Placeholder text. Default 'Add title'.
	 * @param string Replacement title.
```

File: projects/plugins/jetpack/modules/custom-post-types/testimonial.php [hunk 5]
```diff
@@ -373,17 +427,21 @@ function change_default_title( $title ) {
 
 	/**
 	 * Change ‘Title’ column label on all Testimonials page.
+	 *
+	 * @param array $columns An array of column names.
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/custom-post-types/testimonial.php:431]:
```suggestion
	 * @param array $columns An array of column names.
	 * @return array Updated array.
```

File: projects/plugins/jetpack/modules/custom-post-types/testimonial.php [hunk 6]
```diff
@@ -392,43 +450,43 @@ function query_reading_setting( $query ) {
 		}
 	}
 
-	/*
+	/**
 	 * If Infinite Scroll is set to 'click', use our custom reading setting instead of core's `posts_per_page`.
+	 *
+	 * @param array $settings Array of Infinite Scroll settings.
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/custom-post-types/testimonial.php:456]:
```suggestion
	 * @param array $settings Array of Infinite Scroll settings.
	 * @return array Updated `$settings`.
```

File: projects/plugins/jetpack/modules/custom-post-types/testimonial.php [hunk 7]
```diff
@@ -392,43 +450,43 @@ function query_reading_setting( $query ) {
...
+			$settings['posts_per_page'] = get_option( self::OPTION_READING_SETTING, $settings['posts_per_page'] ); // phpcs:ignore WordPress.WP.PostsPerPage.posts_per_page_posts_per_page
 		}
 
 		return $settings;
 	}
 
 	/**
 	 * Add CPT to Dotcom sitemap
+	 *
+	 * @param array $post_types Array of post types included in sitemap.
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/custom-post-types/testimonial.php:471]:
```suggestion
	 * @param array $post_types Array of post types included in sitemap.
	 * @return array Updated `$post_types`.
```

File: projects/plugins/jetpack/modules/custom-post-types/testimonial.php [hunk 8]
```diff
@@ -392,43 +450,43 @@ function query_reading_setting( $query ) {
...
-		$testimonials = wp_count_posts( self::CUSTOM_POST_TYPE );
-		$published_testimonials = $testimonials->publish;
-
-		update_option( self::OPTION_NAME, $published_testimonials );
-	}
-
-	function count_testimonials() {
+	/**
+	 * Count the number of published testimonials.
+	 */
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/custom-post-types/testimonial.php:481]:
```suggestion
	 *
	 * @return int
	 */
```

File: projects/plugins/jetpack/modules/custom-post-types/testimonial.php [hunk 9]
```diff
@@ -503,6 +593,11 @@ function customize_register( $wp_customize ) {
 		}
 	}
 
+	/**
+	 * Add Featured image to theme mod if necessary.
+	 *
+	 * @param array $opt The value of the current theme modification.
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/custom-post-types/testimonial.php:599]:
```suggestion
	 * @param array $opt The value of the current theme modification.
	 * @return array Updated `$opt`.
```

File: projects/plugins/jetpack/modules/custom-post-types/testimonial.php [hunk 10]
```diff
@@ -521,25 +616,35 @@ public function coerce_testimonial_image_to_url( $opt ) {
 	 * Our [testimonial] shortcode.
 	 * Prints Testimonial data styled to look good on *any* theme.
 	 *
+	 * @param array $atts Shortcode attributes.
+	 *
 	 * @return jetpack_testimonial_shortcode_html
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/custom-post-types/testimonial.php:621]:
```suggestion
	 * @return string HTML from `self::jetpack_testimonial_shortcode_html()`.
```

File: projects/plugins/jetpack/modules/custom-post-types/testimonial.php [hunk 11]
```diff
@@ -586,19 +691,21 @@ static function jetpack_testimonial_shortcode( $atts ) {
 	/**
 	 * The Testimonial shortcode loop.
 	 *
+	 * @param array $atts Shortcode attributes.
+	 *
 	 * @return html
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/custom-post-types/testimonial.php:696]:
```suggestion
	 * @return string html
```

File: projects/plugins/jetpack/modules/custom-post-types/testimonial.php [hunk 12]
```diff
@@ -698,17 +809,18 @@ static function get_testimonial_class( $testimonial_index_number, $columns, $ima
...
-		return apply_filters( 'testimonial-entry-post-class', implode( " ", $class ) , $testimonial_index_number, $columns, $image );
+		return apply_filters( 'testimonial-entry-post-class', implode( ' ', $class ), $testimonial_index_number, $columns, $image ); // phpcs:ignore WordPress.NamingConventions.ValidHookName.UseUnderscores
 	}
 
 	/**
 	 * Display the featured image if it's available
 	 *
+	 * @param int $post_id Post ID.
+	 *
 	 * @return html
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/custom-post-types/testimonial.php:821]:
```suggestion
	 * @return string html
```

File: projects/plugins/jetpack/modules/custom-post-types/testimonial.php [hunk 13]
```diff
@@ -724,32 +836,61 @@ static function get_testimonial_thumbnail_link( $post_id ) {
...
+ */
 function jetpack_testimonial_custom_control_classes() {
+	/**
+	 * Clean the title parameter.
+	 */
 	class Jetpack_Testimonial_Title_Control extends WP_Customize_Control {
+		/**
+		 * Sanitize content passed to control.
+		 *
+		 * @param string $value Control value.
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/custom-post-types/testimonial.php:850]:
```suggestion
		 * @param string $value Control value.
		 * @return string Sanitized value.
```

File: projects/plugins/jetpack/modules/custom-post-types/testimonial.php [hunk 14]
```diff
@@ -724,32 +836,61 @@ static function get_testimonial_thumbnail_link( $post_id ) {
...
 				<textarea rows="5" style="width:100%;" <?php $this->link(); ?>><?php echo esc_textarea( $this->value() ); ?></textarea>
 			</label>
-		<?php
+			<?php
 		}
 
+		/**
+		 * Sanitize content passed to control.
+		 *
+		 * @param string $value Control value.
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/custom-post-types/testimonial.php:887]:
```suggestion
		 * @param string $value Control value.
		 * @return string Sanitized value.
```

**Review by anomiex — COMMENTED** [review]
Some doc comment suggestions. Otherwise looks good.

File: projects/plugins/jetpack/modules/custom-post-types/testimonial.php [hunk 15]
```diff
@@ -416,6 +420,7 @@ public function updated_messages( $messages ) {
 	 * Change ‘Enter Title Here’ text for the Testimonial.
 	 *
 	 * @param string $title Placeholder text. Default 'Add title'.
+	 * @param string Replacement title.
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/custom-post-types/testimonial.php:423]: Oops, I made an error in my suggestion
```suggestion
	 * @return string Replacement title.
```

**Review by anomiex — APPROVED** [review]

**Comment by jeherve** [comment]
r244761-wpcom

**Pull request merged**